nisystemlink/clients/file/_file_client.py

```diff
@@ -0,0 +1,253 @@
+        """
+
+    @post("service-groups/Default/delete-files", args=[Body, Query])
+    def delete_files(
+        self, files: models.DeleteMutipleRequest, force: bool = False
+    ) -> None:
+        """Delete multiple files.
+
+        Args:
+            files: The description of files to delete.
+            force: Whether the deletion of files will be forced. Defaults to False.
+
+        Raises:
+            ApiException: if unable to communicate with the File Service.
+        """
+
+    # TBD: Error with poe types - Untyped decorator makes function "download_file" untyped
+    # @params({"inline": True})
+    @response_handler(file_like_response_handler)
+    @get("service-groups/Default/files/{id}/data", args=[Path, Query])
+    def download_file(self, id: str, inline: bool = True) -> IteratorFileLike:
+        """Downloads a file from the SystemLink File service.
+
+        Args:
+            id: The ID of the file.
+            inline: Return the file inline. Defaults to True.
+
+        Yields:
+            A file-like object for reading the exported data.
+
+        Raises:
+            ApiException: if unable to communicate with the File Service.
+        """
+
+    @post("service-groups/Default/upload-files")
+    def __upload_file(
+        self,
+        file: Part,
+        metadata: Part = None,
+        id: Part = None,
+        workspace: Query = None,
+    ) -> models.UploadedFileInfo:
+        """Uploads a file using multipart/form-data headers to send the file payload in the HTTP body.
+
+        Args:
+            file: The file to upload.
+            metadata: JSON Dictionary with key/value pairs
+            id: Specify an unique (among all file) 24-digit Hex string ID of the file once it is uploaded.
+                Defaults to None.
+            workspace: The id of the workspace the file belongs to. Defaults to None.
+
+        Returns:
+            Uploaded file information
+
+        Raises:
+            ApiException: if unable to communicate with the File Service.
+        """
+
+    def upload_file(
+        self,
+        file: BinaryIO,
+        metadata: Optional[Dict[str, str]] = None,
+        id: Optional[str] = None,
+        workspace: Optional[str] = None,
+    ) -> models.UploadedFileInfo:
+        """Uploads a file to the File Service.
+
+        Args:
+            file: The file to upload.
+            metadata: File Metadata as dictionary.
+            id: Specify an unique (among all file) 24-digit Hex string ID of the file once it is uploaded.
+                Defaults to None.
+            workspace: The id of the workspace the file belongs to. Defaults to None.
+
+        Returns:
+            Uploaded file information
+
+        Raises:
+            ApiException: if unable to communicate with the File Service.
+        """
+        if metadata:
+            metadata_str = json.dumps(metadata)
+        else:
+            metadata_str = None
+
+        resp = self.__upload_file(
```

Comment:
Could we parse the URI in the response and just return the string ID from this method?